src/BrightScriptDebugSession.ts

```diff
@@ -287,6 +306,157 @@ export class BrightScriptDebugSession extends DebugSession {
         }
     }
 
+    private componentLibraryPostfix: string = '__lib';
+
+    protected async prepareAndHostComponentLibraries(componentLibraries, componentLibrariesOutDir: string, port: number) {
+        if (componentLibraries && componentLibrariesOutDir) {
+            this.componentLibrariesOutDir = componentLibrariesOutDir;
+            this.componentLibrariesStagingDirPaths = [];
+            let replace = require('replace-in-file');
+            let libraryNumber: number = 0;
+
+            // #region Prepare the component libraries and create some name spacing for debugging
+            for (const componentLibrary of componentLibraries as any) {
+                libraryNumber ++;
+                componentLibrary.outDir = componentLibrariesOutDir;
+                let stagingFolder = await this.rokuDeploy.prepublishToStaging(componentLibrary);
+                let paths = glob.sync(path.join(stagingFolder, '**/*'));
+                let pathDetails: object = {};
+
+                // Add breakpoint lines to the staging files and before publishing
+                this.sendDebugLogLine('Adding stop statements for active breakpoints in Component Libraries');
+                this.convertBreakpointPaths(componentLibrary.rootDir, componentLibrary.rootDir);
+                await this.addBreakpointStatements(stagingFolder, componentLibrary.rootDir);
+
+                for (let filePath of paths) {
```

Comment:
I know it makes this code a little more complicated, but we should consider running each iteration in parallel (not actual parallel, but the node style). So something like this: 

```typescript
await Promise.all(paths.map(async (filePath)=>{
    ....the for loop body
{));
```

This will allow things to complete a bit faster because we are doing work as soon as we are free, instead of iterating through each path in order.